src/widgets/index/index.ts

```diff
@@ -401,6 +496,40 @@ const index = (props: IndexProps): Index => {
       });
 
       derivedHelper.on('search', () => {
+        const pbf = new Pbf();
+        const payload = instantSearchInstance.telemetry.getPayload();
+        Schema.write(payload, pbf);
+        const arrayBuffer = pbf.finish();
+        const newTelemetryHeader = window.btoa(
+          String.fromCharCode.apply(null, arrayBuffer)
+        );
+        // console.log(payload);
+
+        // const mappedPayload = payload.widgets.map(widget => ({
+        //   type: Object.keys(WidgetType).find(
+        //     type => WidgetType[type] === widget.type
+        //   ),
+        //   params: widget.params.map(paramId =>
+        //     Object.keys(WidgetParams).find(
+        //       type => WidgetParams[type] === paramId
+        //     )
+        //   ),
+        //   useConnector: widget.useConnector,
+        // }));
+
+        // console.log(mappedPayload);
+
+        if (telemetryHeader !== newTelemetryHeader) {
+          telemetryHeader = newTelemetryHeader;
+          instantSearchInstance.client.transporter.queryParameters[
```

Comment:
this is only v4 and our client compatible, we will want to send that as a query parameter

Reply:
This part is very hacky here:

* It should be in headers, not in queryParameters
* `instantSearchInstance.client.transporter.queryParameters` is typed at read-only, so to do it properly we'll need to have a proper method to update it

I was not planning to be compatible with the client v3 (since we'll need to do some update to the client to properly set the header). Do you thing this is something we should do?

Reply:
unknown request parameters of the client will go to either headers or query parameters (v4):

```
search([], { someParameter: "value", headers: { someHeader: "value })
```

v3 doesn't have any options for that, but at least it should ignore the second parameter